src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs

```diff
@@ -256,9 +257,15 @@ public string SerializeResult(ExecutionResult result)
                 UserContext = UserContextWrapper.Create(userContext, dependencyInjector ?? new WrappedDependencyInjector(_constructor.TypeResolutionDelegate)),
                 ValidationRules = rules != null && rules.Any() ? rules : null,
                 ComplexityConfiguration = complexityConfiguration,
-                CancellationToken = cancellationToken,
+                CancellationToken = cancellationToken
             };
 
+            if (listeners != null && listeners.Any())
```

Comment:
` && listeners.Any()` is redundant and causes double enumeration. If `listeners` is empty nothing would be added.